ui/src/cloud/apis/demodata.ts

```diff
@@ -0,0 +1,93 @@
+// Libraries
+import {get} from 'lodash'
+import * as api from 'src/client'
+import AJAX from 'src/utils/ajax'
+
+//Utils
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {isDemoData} from 'src/cloud/utils/filterDemoData'
+
+//Types
+import {Bucket} from 'src/types'
+import {LIMIT} from 'src/resources/constants'
+
+const baseURL = '/api/v2/experimental/sampledata'
+
+export const getDemoDataBuckets = async (): Promise<Bucket[]> => {
+  try {
+    const {data} = await AJAX({
+      method: 'GET',
+      url: `${baseURL}/buckets`,
+    })
+
+    // if sampledata endpoints are not available in a cluster
+    // gateway responds with a list of links where 'buckets' field is a string
+    const buckets = get(data, 'buckets', false)
+    if (!buckets || !Array.isArray(buckets)) {
```

Comment:
i *think* you're safe doing this:

```suggestion
    if (!Array.isArray(buckets)) {
```
any falsy value that will pass `!buckets` should also pass `!Array.isArray(buckets)`

Reply:
yup